### ISSUE TITLE
Multiple variable contraint

### ISSUE THREAD

**Issue by NEGU93** [tensorflow/tensorflow#56217]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Feature Request

### Source

source

### Tensorflow Version

2.4.1

### Custom Code

No

### OS Platform and Distribution

Linux Ubuntu

### Python version

3.8.8

### Current Behaviour?

```shell
Based on this [Stackoverflow question](https://stackoverflow.com/questions/68353001/keras-weight-contraints-including-both-bias-and-kernel-matrix-at-the-same-time).

It would be nice to have the possiblilty to apply contraints to multiple variables at the same time. For example, one could apply a maxnorm contraint to a Dense layer that takes into account weigths and biases all together.
```

### Standalone code to reproduce the issue

```shell
There are many UX options here. One possibility (but less scalable for other use cases) is to add the parameter `kernel_bias_contraint` to the Dense layer.
A better one would be to let `tf.keras.constraints.Constraint` to receive multiple variables.
Both options are non exclusive.
```

### Relevant log output

_No response_</details>

**Comment by sushreebarsa** [comment]
@NEGU93  Could you elaborate about your feature and please specify the use cases for this feature. Thank you!

**Comment by NEGU93** [comment]
Sure. 

**Example 1:**
The example of StackOverflow, and, what I meant in the phrase:

> For example, one could apply a maxnorm contraint to a Dense layer that takes into account weigths and biases all together.

Is that maybe a constraint on the norm for weights and biases separately are not enough. Maybe the bias max norm should be lower if the norm of the weigths is already big.

**Example 2**

My use case (so, another example), is that I do a Complex-Valued layer by using real and imaginary parts separately and simulating a complex operation. I want to impose contraint on the norm of the absolute value, which is not possible if I don't know both values at the same time when computing the norm.


Basically, the constraint is applied by knowing or using another variable's value. I would therefore like for the constraint to receive and change two variables at the same time. in the weight/bias example the constraint will receive both `self.w` and `self.b` and perform maxnorm on the concatenation of both values. In the complex-valued case, it will compute the absolute value, perform maxnorm and change real and imag parts to achieve the needed normalized absolute values.

I hope this is clearer.

Regards,

J Agustin BARRACHINA.

**Comment by rchao** [comment]
Hello @NEGU93 , thanks for the request. Would you mind providing usage code snippets of the apis you're proposing so it's clearer what the request is here, for the following two options?

> There are many UX options here. One possibility (but less scalable for other use cases) is to add the parameter `kernel_bias_contraint` to the Dense layer.

> A better one would be to let `tf.keras.constraints.Constraint` to receive multiple variables.
Both options are non exclusive.

**Comment by rchao** [comment]
Giving it some thoughts, does it appear that you'd need both 1) the `Dense` layer to take `kernel_bias_contraint` and 2) `tf.keras.constraints.Constraint` to receive multiple (2 in this case) variables to achieve what you're looking for?

Also, assuming this gets an okay from Keras team to proceed, would you be interested in contributing?

**Comment by rchao** [comment]
Also, is this possible to achieve this with a custom layer?

**Comment by n-j-p** [comment]
> @NEGU93 Could you elaborate about your feature and please specify the use cases for this feature. Thank you!

Hi. I stumbled upon this feature request after finding the stack overflow page referenced above. 

I am trying to fit a (one neuron) simpleRNN layer, and wanted to constrain the absolute sum of the recurrent and kernel weights to be less than 1. If my understanding is correct, this would ensure stationary (i.e. decaying) predictions/projections for a linear activation function.

I can do this by including a regularization parameter, but it would be good to do this with a constraint. As far as I am aware, this is currently impossible/difficult as mentioned above since constraints work on recurrent and kernels separately.

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
